File: Typescript/inorderTraversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function inorderTraversal(root: TreeNode | null): number[] {
    const result: number[] = [];
    const stack: TreeNode[] = [];
    let tmp: TreeNode | null = root;
    
    while (tmp || stack.length) {
        if (tmp) {
            stack.push(tmp);
            tmp = tmp.left;
        } else {
            tmp = stack.pop()!;
            result.push(tmp.val);
            tmp = tmp.right;
        }
    }
    
    return result;
};

File: C++/inorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) 
    {
        vector<int> result;
        stack<TreeNode*> s;
        TreeNode* tmp = root;
        
        while (tmp || !s.empty()) 
        {
            if (tmp) 
            {
                s.push(tmp);
                tmp = tmp->left;
            } 
            else 
            {
                tmp = s.top();
                s.pop();
                result.push_back(tmp->val);
                tmp = tmp->right;
            }
        }
        
        return result;
    }
};